test/methods.jl

```diff
@@ -11,3 +11,14 @@
     # testing error
     @test_throws ErrorException bias_subtraction(ones(500, 1), ones(500, 500))
 end
+
+@testset "overscan subtraction" begin
+    # testing non-mutating version
+    @test @inferred overscan_subtraction(ones(500, 600), ones(2, 600), overscan_axis = 1) == zeros(500, 600)
```

Comment:
the `@inferred` macro here is testing if the expression `... == ...` is type-stable, not the left-hand side. You need to do `@inferred(expr) == ...`

Reply:
Had to remove `@inferred` as the current version of code is not type-stable.
I modified the tests for `bias_subtraction` it had the same error and was not testing type-stability properly.